src/elements/List/List.js

```diff
@@ -1,32 +1,102 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
 import META from '../../utils/Meta'
+import * as sui from '../../utils/semanticUtils'
+import {
+  getUnhandledProps,
+  useValueAndKey,
+  useKeyOrValueAndKey,
+  useKeyOnly,
+} from '../../utils/propUtils'
 import ListItem from './ListItem'
 
-export default class List extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-  }
-
-  static _meta = {
-    library: META.library.semanticUI,
-    name: 'List',
-    type: META.type.element,
-  }
-
-  static Item = ListItem
-
-  render() {
-    const classes = classNames(
-      'ui',
-      this.props.className,
-      'list'
-    )
-    return (
-      <div {...this.props} className={classes}>
-        {this.props.children}
-      </div>
-    )
-  }
+function List(props) {
+  const {
+    size, aligned, bulleted, link, ordered, animated, celled, divided,
+    horizontal, inverted, relaxed, selection, children, className,
+    } = props
+
+  const classes = cx(
+    'ui',
+    size,
+    useValueAndKey(aligned, 'aligned'),
+    useKeyOnly(bulleted, 'bulleted'),
+    useKeyOnly(link, 'link'),
```

Comment:
When the [`link`](http://semantic-ui.com/elements/list.html#link) class is used, the `ListItem`s need to use an `a` tag:

```html
<div class="ui link list">
  <div class="active item">Home</div>
  <a class="item">About</a>
  <a class="item">Jobs</a>
  <a class="item">Team</a>
</div>
```

I've considered adding a `component` prop to every component, which would allow setting the component to render as.  In this case, an `a`.  When a `List` has `link` set, we could map all the `ListItem` children and set it like so `<List.Item component='a' />`.

Note the `active` item always uses a `div`.  So, a `<List.Item active />` should always render a `div`.

Thoughts?